libraries/ms-common/test/test_x86_uart.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import socket
+import _thread
+import time
+
+# return the sun_path of the most recently started program
+def get_sun_path(prog, port):
+
+    latest_path = ""
+    res = subprocess.run(["pgrep", "-l", prog], stdout = subprocess.PIPE)
+    # decode pgrep output from bytes and get the pids and program names
+    res_arr = res.stdout.decode("utf-8").split("\n")
+
+    for proc in res_arr:
+        if proc:
+            try:
+                tmp_arr = proc.split(" ")
+                if tmp_arr[1] == prog:
+                    latest_path = "%s/%s/%s" % (tmp_arr[0],prog,port)
+            except:
+                continue
+
+    return latest_path
```

Comment:
This will always return the last result that pgrep returns. If the goal is to use the most recently launched program, you could use pgrep's `-n` arg.

Reply:
It would be very useful (especially because then the python script could connect different instances of the same program). I guess if the user started each of the programs they intended to connect (the PID of the server is logged on startup) and ran the python script after this then they could specify the pid and program name of each program.

Reply:
I think I'll implement that as an optional feature

Reply:
I would advise using argparse for command line parsing. The ability to load a config file would be nice.